tests/experimental/nn/test_sync_batchnorm.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import functools
+import tempfile
+
+import pytest
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from fairscale.experimental.nn import SyncBatchNorm
+
+pytestmark = pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
+
+
+def pg_worker(rank, world_size, init_file, func, *args):
+    dist.init_process_group(dist.Backend.NCCL, init_method="file://" + init_file, rank=rank, world_size=world_size)
```

Comment:
Since this is a fundamental module should we test the GLOO backend as well? if things don't work as expected we can then limit to NCCL support but would help surface any GLOO specific issues.

Reply:
Cool if I handle that in a follow-up change since its marked as experimental? Wanted to unblock testing of the code by others.

Reply:
Absolutely! thanks for following up.